app/src/main/java/com/github/onlynotesswent/model/note/Note.kt

```diff
@@ -65,4 +67,96 @@ data class Note(
+        val timestamp = System.currentTimeMillis()
+        val input = "$userId$content$timestamp"
+        val digest = MessageDigest.getInstance("SHA-256")
+        val hashBytes = digest.digest(input.toByteArray())
+        return hashBytes.joinToString("") { "%02x".format(it) }
+      }
+      /**
+       * Adds a new comment to the CommentCollection with a unique ID.
+       *
+       * @param userId The ID of the user adding the comment.
+       * @param content The text content of the comment.
+       * @param CommentCollection The current collection of comments.
+       * @return An updated CommentCollection including the new comment.
+       */
+      fun addComment(
+          userId: String,
+          content: String,
+          CommentCollection: CommentCollection
+      ): CommentCollection {
+        val mutableCommentsList = CommentCollection.commentsList.toMutableList()
+        mutableCommentsList.add(0, Comment(generateId(userId, content), userId, content))
+        return CommentCollection(mutableCommentsList.toList())
+      }
+      /**
+       * Edits an existing comment in the CommentCollection by its ID.
+       *
+       * @param commentId The ID of the comment to be edited.
+       * @param content The new content for the comment.
+       * @param CommentCollection The current collection of comments.
+       * @return An updated CommentCollection with the modified comment.
+       */
+      fun editComment(
+          commentId: String,
+          content: String,
+          CommentCollection: CommentCollection
+      ): CommentCollection {
+        val mutableCommentsList = CommentCollection.commentsList.toMutableList()
+        return CommentCollection(
+            mutableCommentsList.map {
+              if (it.commentId == commentId)
+                  Comment(generateId(it.userId, content), it.userId, content)
+              else it
+            })
+      }
+      /**
+       * Deletes a comment from the CommentCollection by its ID.
+       *
+       * @param commentId The ID of the comment to be deleted.
+       * @param CommentCollection The current collection of comments.
+       * @return An updated CommentCollection excluding the deleted comment.
+       */
+      fun deleteComment(
+          commentId: String,
+          CommentCollection: CommentCollection
+      ): CommentCollection {
+        return CommentCollection(
+            CommentCollection.commentsList.filter { it.commentId != commentId })
+      }
+    }
+  }
+
+  /**
+   * Represents a comment that is stored in the CommentCollection.
+   *
+   * @property commentId A unique identifier for this comment, generated by hashing the user ID,
+   *   content, and timestamp.
+   * @property userId The unique identifier of the user who posted the comment.
+   * @property content The text content of the comment.
+   */
+  data class Comment(val commentId: String, val userId: String, val content: String)
```

Comment:
For more readability it is better to put the arguments of `Comment` in column format